core/src/util/builders.cpp

```diff
@@ -383,7 +372,7 @@ void Builders::buildPolyLine(const Line& _line, PolyLineBuilder& _ctx) {
         for (size_t i = 0; i < _line.size() - 1; i++) {
             const glm::vec3& coordCurr = _line[i];
             const glm::vec3& coordNext = _line[i+1];
-            if (isOnTileEdge(coordCurr, coordNext)) {
+            if (isOutsideTile(coordCurr, coordNext)) {
                 Line line = Line(&_line[cut], &_line[i+1]);
                 buildPolyLineSegment(line, _ctx);
```

Comment:
@hjanetzek I am not sure what do you mean by `&_line[cut + length]`, if its that we can ignore all the coordinates past "this" edge which is outside the tile bounds then I do not think we should do that. Its possible some edges between `i+1` and `length` are within the tile bounds. [by edges I mean line segments formed by the line coordinates].

And `length == 1` will automatically be checked in `buildPolyLineSegment`.

I agree on extending the tile bbox to do the bound check. Will change this.